components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -301,7 +301,8 @@ RewardsServiceImpl::RewardsServiceImpl(Profile* profile)
       private_observer_(
           std::make_unique<ExtensionRewardsServiceObserver>(profile_)),
 #endif
-      next_timer_id_(0) {
+      next_timer_id_(0),
+      weak_ptr_factory_(this) {
```

Comment:
I think I had a comment that got lost, but `weak_ptr_factory_` isn't necessary. It already subclasses SupportsWeakPtr so you can just use `AsWeakPtr` like the others

Reply:
Fixed